src/Microsoft.Framework.Primitives/StringValues.cs

```diff
@@ -114,28 +115,64 @@ private string[] GetArrayValue()
 
         int IList<string>.IndexOf(string item)
         {
-            var index = 0;
-            foreach (var value in this)
+            return IndexOf(item);
+        }
+
+        private int IndexOf(string item)
+        {
+            if (_values != null)
             {
-                if (string.Equals(value, item, StringComparison.Ordinal))
+                string[] values = _values;
```

Comment:
The JIT is likely to produce better code when its cached in a local. See comment by jkotas: https://github.com/dotnet/coreclr/pull/1231#discussion_r34603505